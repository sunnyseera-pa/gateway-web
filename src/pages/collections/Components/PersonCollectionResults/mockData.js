export const data = [
	{
		_id: '5e544facbd427b6e9cd9059e',
		id: 900000011,
		lastname: 'Nei',
		bio: 'Jefferson Comprehensive Epilepsy Center',
		firstname: 'Maromi',
		type: 'person',
		tags: ['Neuroendocrinology', 'Epilepsy', 'Neuroscience'],
		activeflag: 'active',
		authors: [''],
		counter: 75,
		relatedresources: 0,
		persons: [],
		updated: 1621551600000,
	},
];

export const relatedObjects = [
	{
		_id: '60a7bba5bfd44334afc2d069',
		objectId: '900000011',
		reason: '',
		objectType: 'person',
		pid: '',
		user: 'Ciara Ward',
		updated: '21 May 2021',
	},
];
