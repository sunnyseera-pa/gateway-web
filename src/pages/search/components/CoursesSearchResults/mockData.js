export const data = [
    {
        id: 50033,
        award: [],
        courseOptions: {
            flexibleDates: true,
            startDate: null,
            studyMode: '',
        },
        creator: 50000,
        description:
            'Health data analytics involves extracting insights from health data, either to shape national policy, manage local organisations or inform the care of an individual. As more and more data becomes available electronically, the demand for skilled and trained individuals to take advantage of it becomes increasingly urgent.',
        domains: ['Course'],
        provider: 'UCL Institute of Health Informatics',
        title: 'Health Data Analytics ',
        type: 'course',
        myEntity: false,
    },
    {
        id: 50017,
        award: [],
        courseOptions: {
            flexibleDates: true,
            startDate: null,
            studyMode: '',
        },
        creator: 50000,
        description:
            'The aim of this course is to familiarize the participants with the primary analysis of RNA-seq data.\n\nThis course starts with a brief introduction to RNA-seq and discusses quality control issues. Next, we will present the alignment step, quantification of expression and differential expression analysis. For downstream analysis we will focus on tools available through the Bioconductor project for manipulating and analysing bulk RNA-seq.',
        domains: ['Course'],
        provider: 'University of Cambridge',
        title: 'Introduction to RNA-seq data analysis',
        type: 'course',
        myEntity: false,
    },
    {
        id: 50005,
        award: [],
        courseOptions: {
            flexibleDates: true,
            startDate: null,
            studyMode: '',
        },
        creator: 50000,
        description:
            'This course shows you how to do the basics in R from reading and writing to files, manipulating data, and visualising it by creating different plots. During the course you will work with the R-package: tidyverse.',
        domains: ['Course'],
        provider: 'University of Cambridge',
        title: 'Introduction to R for Biologists',
        type: 'course',
        myEntity: false,
    },
    {
        id: 50029,
        award: [],
        courseOptions: {
            flexibleDates: true,
            startDate: null,
            studyMode: '',
        },
        creator: 50000,
        description:
            'Our one day breast cancer bioinformatics course, funded by Breast Cancer Now, offers an introduction to the use of bioinformatics. The course offers a unique opportunity for breast cancer biologists and clinicians to gain a basic understanding of the use of bioinformatics methods to access and harness large complicated high-throughput data and uncover meaningful information that could be used to understand molecular mechanisms and develop novel targeted therapeutics and diagnostic tools.  \n\n',
        domains: ['Course'],
        provider: 'Barts Cancer Research Institute',
        title: 'Breast Cancer Bioinformatics Course',
        type: 'course',
        myEntity: false,
    },
    {
        id: 50021,
        award: [],
        courseOptions: {
            flexibleDates: true,
            startDate: null,
            studyMode: '',
        },
        creator: 50000,
        description:
            'Executive master course (MBA) providing part-time training over 18 months. Modules are also available on a standalone basis, and webinars are being made openly available',
        domains: ['Course'],
        provider:
            'RItrain consortium, which includes EMBL-EBI; project lead is BBMRI (currently hub and BBMRI Italy but moving to BBMRI Italy, based in Milan)',
        title: "RItrain's Executive Master in Management of Research Infrastructures",
        type: 'course',
        myEntity: false,
    },
    {
        id: 50013,
        award: [],
        courseOptions: {
            flexibleDates: true,
            startDate: null,
            studyMode: '',
        },
        creator: 50000,
        description:
            'This course will present a set of R/Bioconductor packages to access, manipulate, visualise and analyse mass spectrometry (MS) and quantitative proteomics data.',
        domains: ['Course'],
        provider: 'University of Cambridge',
        title: 'Exploring, visualising and analysing proteomics data in R',
        type: 'course',
        myEntity: false,
    },
    {
        id: 50015,
        award: [],
        courseOptions: {
            flexibleDates: true,
            startDate: null,
            studyMode: '',
        },
        creator: 50000,
        description:
            'Machine learning gives computers the ability to learn without being explicitly programmed. It encompasses a broad range of approaches to data analysis with applicability across the biological sciences. Lectures will introduce commonly used algorithms and provide insight into their theoretical underpinnings. In the practicals students will apply these algorithms to real biological data-sets using the R language and environment.',
        domains: ['Course'],
        provider: 'University of Cambridge',
        title: 'Introduction to Machine Learning',
        type: 'course',
        myEntity: false,
    },
    {
        domains: ['Course'],
        award: [],
        courseOptions: {
            flexibleDates: true,
            startDate: null,
            studyMode: 'Full-time',
        },
        id: 1464780631562841,
        type: 'course',
        creator: 5703995672678985,
        title: 'Safe Researcher Training',
        provider: 'UK Data Service',
        description:
            'If you want to become an Accredited Researcher under the Digital Economy Act then you will need to complete recognised Safe Researcher Training. DEA accreditation is a requirement for access to the microdata contained in the ONS Secure Research Service, and also is recognised by SAIL (Wales), EPCC (Scotland) and HBS (NI) TREs.\n\nOn this course you’ll gain an understanding of the different types of Statistical Disclosure and learn about different statistical techniques to deal with them.  You will learn the importance of protecting confidentiality and the about the 5-safes framework, many data providers use to keep data safe.  \n\nTo gain access to microdata in the ONS SRS you will also need a project application approved separately by the Research Accreditation Panel. \n\nThis course is provided regularly online by the UK Data Service. Course Duration: 4 hour online\n\nEntry Requirements: an undergraduate degree (or higher) including a significant proportion of mathematics or statistics; or, be able to demonstrate at least 3 years quantitative research experience.',
        myEntity: false,
    },
    {
        id: 50019,
        award: [],
        courseOptions: {
            flexibleDates: true,
            startDate: null,
            studyMode: '',
        },
        creator: 50000,
        description:
            'R is a highly-regarded, free, software environment for statistical analysis, with many useful features that promote and facilitate reproducible research.\n\nIn this course, we give an introduction to the R environment and explain how it can be used to import, manipulate and analyse tabular data. After the course you should feel confident to start exploring your own dataset using the materials and references provided.',
        domains: ['Course'],
        provider: 'University of Cambridge',
        title: 'Introduction to solving biological problems in R',
        type: 'course',
        myEntity: false,
    },
    {
        id: 50007,
        award: [],
        courseOptions: {
            flexibleDates: true,
            startDate: null,
            studyMode: '',
        },
        creator: 50000,
        description:
            'Partners in the Cambridge MRC DTP share a mission to contribute to society through the pursuit of education, learning and research at the highest international levels of excellence.',
        domains: ['Course'],
        provider: 'School of Clinical Medicine, The Babraham Institute and School of Biological Sciences',
        title: 'DTP Graduate training (MRC)',
        type: 'course',
        myEntity: false,
    },
    {
        id: 50027,
        award: [],
        courseOptions: {
            flexibleDates: true,
            startDate: null,
            studyMode: '',
        },
        creator: 50000,
        description: '',
        domains: ['Course'],
        provider: 'Wellcome Sanger Institute',
        title: 'Wellcome Sanger 4-year PhD programme',
        type: 'course',
        myEntity: false,
    },
    {
        id: 50023,
        award: [],
        courseOptions: {
            flexibleDates: true,
            startDate: null,
            studyMode: '',
        },
        creator: 50000,
        description:
            'ELIXIR is an intergovernmental organisation that brings together life science resources from across Europe. Its Training portal, TeSS, allows you to browse and discover life sciences training resources, aggregated from ELIXIR nodes and 3rd-party providers.',
        domains: ['Course'],
        provider: 'ELIXIR',
        title: 'TeSS, ELIXIR training portal',
        type: 'course',
        myEntity: false,
    },
    {
        id: 50011,
        award: [],
        courseOptions: {
            flexibleDates: true,
            startDate: null,
            studyMode: '',
        },
        creator: 50000,
        description:
            'Regular quarterly strategy meetings, expert-level workshops on topics prioritised by the members, webinars and other activities. It also serves as an interface between industry-focussed initiatives at EMBL-EBI and organisations such as the Innovative Medicines Initiative (IMI), the Pistoia Alliance, the Clinical Data Interchange Standards Consortium (CDISC) and others.',
        domains: ['Course'],
        provider: 'EMBL-EBI',
        title: 'EMBL-EBI Industry programme',
        type: 'course',
        myEntity: false,
    },
    {
        id: 50031,
        award: [],
        courseOptions: {
            flexibleDates: true,
            startDate: null,
            studyMode: '',
        },
        creator: 50000,
        description:
            'A series of one and two day short courses aimed at researchers in health data science, dealing with a mix of technical skills (R, python, SQL) and methods (mendelian randomisation, data visualisation, machine learning, phenotyping)',
        domains: ['Course'],
        provider: 'Institute of Health Informatics/UCL',
        title: 'Harnessing Electronic Health Records for Research',
        type: 'course',
        myEntity: false,
    },
    {
        id: 50003,
        award: [],
        courseOptions: {
            flexibleDates: true,
            startDate: null,
            studyMode: '',
        },
        creator: 50000,
        description:
            'The primary aim of this course is to familiarise participants with the analysis of ChIP-seq and ATAC-seq data and provide hands-on training on the latest analytical approaches.\nThe course starts with an introduction to ChIP-seq experiments for the detection of genome-wide DNA binding sites of transcription factors and other proteins. We first show data quality control and basic analytical steps such as alignment, peak calling and motif analysis, followed by practical examples on how to work with biological replicates and fundamental quality metrics for ChIP-seq datasets. On the second day, we then focus on the analysis of differential binding, comparing between different samples. We will also give an introduction to ATAC-seq data analysis for the detection of regions of open chromatin.',
        domains: ['Course'],
        provider: 'University of Cambridge',
        title: 'ChIP-seq and ATAC-seq analysis',
        type: 'course',
        myEntity: false,
    },
    {
        id: 50034,
        award: [],
        courseOptions: {
            flexibleDates: true,
            startDate: null,
            studyMode: '',
        },
        creator: 50000,
        description:
            'Health data analytics involves extracting insights from health data, either to shape national policy, manage local organisations or inform the care of an individual. As more and more data becomes available electronically, the demand for skilled and trained individuals to take advantage of it becomes increasingly urgent.',
        domains: ['Course'],
        provider: 'UCL Institute of Health Informatics',
        title: 'Health Data Analytics ',
        type: 'course',
        myEntity: false,
    },
    {
        id: 50004,
        award: [],
        courseOptions: {
            flexibleDates: true,
            startDate: null,
            studyMode: '',
        },
        creator: 50000,
        description:
            'In many domains of research the rapid generation of large amounts of data is fundamentally changing how research is done. The deluge of data presents great opportunities, but also many challenges in managing, analyzing and sharing data.\n\nData Carpentry workshops are designed to teach basic concepts, skills and tools for working more effectively with data, using a combination of tools with a main focus in R. The workshop is aimed at researchers in the life sciences at all career stages and is designed for learners with little to no prior knowledge of programming, shell scripting, or command line tools.',
        domains: ['Course'],
        provider: 'University of Cambridge',
        title: 'Data Carpentry in R',
        type: 'course',
        myEntity: false,
    },
    {
        id: 50016,
        award: [],
        courseOptions: {
            flexibleDates: true,
            startDate: null,
            studyMode: '',
        },
        creator: 50000,
        description:
            'The goal of metabolomics is to identify and quantify the complete biochemical composition of a biological sample. With the increase in genomic, transcriptomic and proteomic information there is a growing need to understand the metabolic phenotype that these genes and proteins ultimately control.\n\nThe aim of this course is to provide an overview of metabolomics and its applications in life sciences, clinical and environmental settings. Over 2 days we will introduce different techniques used to extract metabolites and analyse samples to collect metabolomic data (such as HPLC or GC-based MS and NMR), present how to analyse such data, how to identify metabolites using online databases and how to map the metabolomic data to metabolic pathways.\n\nThe course content will predominantly be based on analysing samples from model plant species such as Arabidopsis thaliana but the procedures are transferable to all other organisms, including clinical and environmental settings.',
        domains: ['Course'],
        provider: 'University of Cambridge',
        title: 'Introduction to metabolomics and its application in life-sciences',
        type: 'course',
        myEntity: false,
    },
    {
        domains: ['Course'],
        award: [],
        courseOptions: {
            flexibleDates: true,
            startDate: null,
            studyMode: '',
        },
        id: 4703662242387352,
        type: 'course',
        creator: 5703995672678985,
        title: 'NHS Data Security Awareness',
        provider: 'NHS Health Education England',
        description:
            'An eLearning course from Health Education England to teach researchers the basics of data security and using data safely.\n\nIt is split into four 12 minute modules with an assessment at the end of each module. Users must pass each assessment with 80% or more and a certificate will be issued when all modules are completed and assessments passed.\n\nModules include:\n\n• Module 1 - Introduction to Data Security Awareness • Module 2 - Introduction to the law and how it works with the NHS and care sector • Module 3 - Threats to Data Security and how to avoid them • Module 4- Data Breaches\n\nResearchers must complete this course for research using data held in the NHS Digitial Trusted Research Environment.',
        myEntity: false,
    },
    {
        id: 50018,
        award: [],
        courseOptions: {
            flexibleDates: true,
            startDate: null,
            studyMode: '',
        },
        creator: 50000,
        description:
            'This course provides a practical introduction to Python programming language for the complete novice.\n\nParticipants are lead through the core aspects of Python illustrated by a series of example programs. Upon completion of the course, attentive participants will be able to write simple Python programs from scratch and to customize more complex code to fit their needs.',
        domains: ['Course'],
        provider: 'University of Cambridge',
        title: 'Introduction to solving biological problems in Python',
        type: 'course',
        myEntity: false,
    },
    {
        id: 50006,
        award: [],
        courseOptions: {
            flexibleDates: true,
            startDate: null,
            studyMode: '',
        },
        creator: 50000,
        description:
            'This course covers concepts and strategies for working more effectively with Python with the aim of writing reusable code, using function and libraries. Participants will acquire a working knowledge of key concepts which are prerequisites for advanced programming in Python e.g. writing modules and classes.\n\nNote: this course is the continuation of the Introduction to Solving Biological Problems with Python; participants are expected to have attended the introductory Python course and/or have acquired some working knowledge of Python. This course is also open to Python beginners who are already fluent in other programming languages as this will help them to quickly get started in Python.',
        domains: ['Course'],
        provider: 'University of Cambridge',
        title: 'Data Science in Python',
        type: 'course',
        myEntity: false,
    },
    {
        id: 50020,
        award: [],
        courseOptions: {
            flexibleDates: true,
            startDate: null,
            studyMode: '',
        },
        creator: 50000,
        description:
            'The MRC Biostatistics Unit run a number of successful courses in statistics on a range of topics at different levels, for statistical, clinical and other audiences. ',
        domains: ['Course'],
        provider: 'University of Cambridge',
        title: 'MRC Biostatistics short courses',
        type: 'course',
        myEntity: false,
    },
    {
        id: 50000,
        award: [],
        courseOptions: {
            flexibleDates: true,
            startDate: null,
            studyMode: '',
        },
        creator: 50000,
        description:
            'Recent technological advances have made it possible to obtain genome-wide transcriptome data from single cells using high-throughput sequencing (scRNA-seq). Even though scRNA-seq makes it possible to address problems that are intractable with bulk RNA-seq data, analysing scRNA-seq is also more challenging.\nIn this course we will be surveying the existing problems as well as the available computational and statistical frameworks available for the analysis of scRNA-seq.\n\n',
        domains: ['Course'],
        provider: 'University of Cambridge',
        title: 'Analysis of single-cell RNA-seq data',
        type: 'course',
        myEntity: false,
    },
    {
        id: 50022,
        award: [],
        courseOptions: {
            flexibleDates: true,
            startDate: null,
            studyMode: '',
        },
        creator: 50000,
        description:
            'This course is intended to provide a strong foundation in practical statistics and data analysis using the R software environment. The underlying philosophy of the course is to treat statistics as a practical skill rather than as a theoretical subject and as such the course focuses on methods for addressing real-life issues in the biological sciences using the R software package.\n\nIn this course we explore classical statistical analysis techniques starting with simple hypothesis testing and building up to multiple linear regression. The focus of the course is on practical implementation of these techniques and developing robust statistical analysis skills rather than on the underlying statistical theory.\n\nAfter the course you should feel confident to be able to select and implement common statistical techniques using R and moreover know when, and when not, to apply these techniques.',
        domains: ['Course'],
        provider: 'University of Cambridge',
        title: 'Statistics for Biologists in R',
        type: 'course',
        myEntity: false,
    },
    {
        id: 50008,
        award: ['PhD'],
        courseOptions: {
            flexibleDates: true,
            startDate: null,
            studyMode: '',
        },
        creator: 50000,
        description:
            'EIPOD projects capitalise on synergies emerging at the borders of overlapping yet separate scientific fields, and involve transferring techniques to a novel context. The programme is open to all nationalities and offers about 20 positions annually.',
        domains: ['Course'],
        provider: 'EMBL (including EMBL-EBI)',
        title: 'EMBL EI3POD programme',
        type: 'course',
        myEntity: false,
    },
    {
        id: 50026,
        award: ['Bachelors'],
        courseOptions: {
            flexibleDates: true,
            startDate: null,
            studyMode: '',
        },
        creator: 50000,
        description:
            'Short courses (combinarion of lab-based, computational and lecture/discussion) on the latest biomedical techniques for research scientists, clinicans and healthcare professionals',
        domains: ['Course'],
        provider: 'Connecting Science',
        title: 'Wellcome Genome Campus Advanced Courses',
        type: 'course',
        myEntity: false,
    },
    {
        id: 50024,
        award: [],
        courseOptions: {
            flexibleDates: true,
            startDate: null,
            studyMode: '',
        },
        creator: 50000,
        description:
            'Microscopy experiments have proven to be a powerful means of generating information-rich data for biological applications. From small-scale microscopy experiments to time-lapse movies and high-throughput screens, automatic image analysis is more objective and quantitative and less tedious than visual inspection.\n\nThis course will introduce users to the free open-source image analysis program CellProfiler and its companion data exploration program CellProfiler Analyst. We will show how CellProfiler can be used to analyse a variety of types of imaging experiments. We will also briefly discuss the basic principles of supervised machine learning with CellProfiler Analyst in order to score complex and subtle phenotypes.\n\n',
        domains: ['Course'],
        provider: 'University of Cambridge',
        title: 'Using CellProfiler and CellProfiler Analyst to analyse biological images',
        type: 'course',
        myEntity: false,
    },
    {
        id: 50010,
        award: ['PhD'],
        courseOptions: {
            flexibleDates: true,
            startDate: null,
            studyMode: '',
        },
        creator: 50000,
        description: 'Projects that combine experimental (wet-lab) and computational (dry-lab) approaches.',
        domains: ['Course'],
        provider: 'EMBL-EBI and Sanger',
        title: 'EMBL-EBI ESPOD programme',
        type: 'course',
        myEntity: false,
    },
    {
        id: 50028,
        award: [],
        courseOptions: {
            flexibleDates: true,
            startDate: null,
            studyMode: '',
        },
        creator: 50000,
        description: 'Medically qualified individuals looking to pursue a career in academic medicine',
        domains: ['Course'],
        provider: 'Wellcome Sanger Institute in collaboration with the Cambridge Wellcome PhD programme for clinicians',
        title: 'Wellcome Sanger Clincial PhDprogramme',
        type: 'course',
        myEntity: false,
    },
    {
        id: 50002,
        award: [],
        courseOptions: {
            flexibleDates: true,
            startDate: null,
            studyMode: '',
        },
        creator: 50000,
        description:
            'Broad range of undergraduate and postgraduate hands-on training courses focused on bioinformatics and data science. These training activities aim at enabling life scientists to effectively handle and interpret biological and biomedical data.',
        domains: ['Course'],
        provider: 'University of Cambridge',
        title: 'Bioinformatics Training Program',
        type: 'course',
        myEntity: false,
    },
    {
        id: 50030,
        award: [],
        courseOptions: {
            flexibleDates: true,
            startDate: null,
            studyMode: '',
        },
        creator: 50000,
        description:
            'This specialist workshop on GIS for Public Health offers an introduction to GIS and how it is used in public health and epidemiological research. You will learn how to use GIS to prepare and analyse data for studies of environment and health, use GIS to assess exposures to environmental pollution, and explore and analyse patterns and spread of disease.\nSpecific skills and tools will be introduced in relation to assessing exposure to a range of environmental risks, and a particular focus will be given to methods for linking geographical and health data for epidemiological studies and health risk assessment.',
        domains: ['Course'],
        provider: 'Imperial College London, School of Public Health. Utrecht University, Institute for risk assessment sciences',
        title: 'GIS for Public Health',
        type: 'course',
        myEntity: false,
    },
    {
        id: 50012,
        award: ['Bachelors'],
        courseOptions: {
            flexibleDates: true,
            startDate: null,
            studyMode: '',
        },
        creator: 50000,
        description:
            'Short bioinformatics courses at EMBL-EBI, at host sites and online empowering researchers to make the most of molecular biodata',
        domains: ['Course'],
        provider: 'EMBL-EBI',
        title: 'EMBL-EBI training programme',
        type: 'course',
        myEntity: false,
    },
    {
        id: 50032,
        award: [],
        courseOptions: {
            flexibleDates: true,
            startDate: null,
            studyMode: '',
        },
        creator: 50000,
        description:
            'Health data analytics involves extracting insights from health data, either to shape national policy, manage local organisations or inform the care of an individual. As more and more data becomes available electronically, the demand for skilled and trained individuals to take advantage of it becomes increasingly urgent.',
        domains: ['Course'],
        provider: 'UCL Institute of Health Informatics',
        title: 'Health Data Analytics ',
        type: 'course',
        myEntity: false,
    },
    {
        domains: ['Course'],
        award: [],
        courseOptions: {
            flexibleDates: true,
            startDate: null,
            studyMode: 'Full-time',
        },
        id: 9952579202069596,
        type: 'course',
        creator: 5703995672678985,
        title: 'MRC Regulatory Support Centre: Research Data and Confidentiality e-learning - DISCONTINUED',
        provider: 'Medical Research Council',
        description:
            '** THIS COURSE HAS BEEN DISCONTINUED**\n\nThis MRC course has been discontinued but is still accepted as approved training for researchers to conduct research in the SAIL and EPCC Scotland Trust Research Environments if they have completed it previously.  \n\nCourse Duration: 3 hours online\n\nMRC provides an alternative course that is recognized by SAIL and EPCC Scotland: https://web.www.healthdatagateway.org/course/7853810507803276',
        myEntity: false,
    },
    {
        id: 50014,
        award: [],
        courseOptions: {
            flexibleDates: true,
            startDate: null,
            studyMode: '',
        },
        creator: 50000,
        description:
            'This course will focus on computational methods for analysing cellular images and extracting quantitative data from them. The aim of this course is to familiarise the participants with computational image analysis methodologies, and to provide hands-on training in running quantitative analysis pipelines.\nOn day 1 we will introduce principles of image processing and analysis, giving an overview of commonly used algorithms through a series of talks and practicals based on Fiji, an extensible open source software package.\nOn day 2, we will cover time series processing and cell tracking using TrackMate and advanced image segmentation using Ilastik. Additionally, in the afternoon we will run a study design and data clinic (sign up will be required) for participants that wish to discuss their experiments.\nOn day 3, we will describe the open Icy platform developed at the Institut Pasteur. Icy is a next-generation, user-friendly software offering powerful acquisition, visualisation, annotation and analysis algorithms for 5D bioimaging data, together with unique automation/scripting capabilities (notably via its graphical programming interface) and tight integration with existing software (e.g. ImageJ, Matlab, Micro-Manager).',
        domains: ['Course'],
        provider: 'University of Cambridge',
        title: 'Image analysis for biologists',
        type: 'course',
        myEntity: false,
    },
    {
        domains: ['Courses'],
        award: [],
        courseOptions: {
            flexibleDates: true,
            startDate: null,
            studyMode: 'Full-time',
        },
        id: 7401849492159451,
        type: 'course',
        creator: 5703995672678985,
        title: 'Safe Researcher Training',
        provider: 'Administrative Data Research Centre Scotland',
        description:
            'An in-person one day course delivered by ADRS-Scotland in Edinburgh.\n\nThere are currently no dates set for the course, but if you are interested in taking this course please email adrc-s@ed.ac.uk for more information.\n\nThis course is recognized by EPCC Scotland.',
        myEntity: false,
    },
    {
        domains: ['Course'],
        award: [],
        courseOptions: {
            flexibleDates: true,
            startDate: null,
            studyMode: 'Part-time',
        },
        id: 7853810507803276,
        type: 'course',
        creator: 5703995672678985,
        title: 'Research, GDPR and Confidentiality - what you really need to know',
        provider: 'Medical Research Council',
        description:
            "An eLearning course from MRC introducing MRC's expectations for good research practice. It is a series of 10 bite-sized e-learning modules accompanied by a collection of resources to help researchers get to grips with the principles of GDPR and the common law of confidentiality for research.\n\nModules include:\n\n* What has not changed with GDPR? \n* What makes information 'identifiable'?\n* GDPR, Personal Data and Data Controllers\n* GDPR, Personal Data and Data Controllers\n* GDPR principles - being lawful, fair and transparent\n* Safeguards, archiving and secondary uses of personal data\n* Sharing Data\n* GDPR - know your rights\n* GDPR and accessing information without consent (including England's National Opt-Out)\n* GDPR and accessing information without consent (Scotland)\n* Bringing it all together -- some myths and help with workshop scenarios\n\nYou must create an account to complete the training.\n\nResearchers must complete this course for research using data held in SAIL, EPCC Scotland, and HBS Northern Ireland.",
        myEntity: false,
    },
    {
        id: 50001,
        award: [],
        courseOptions: {
            flexibleDates: true,
            startDate: null,
            studyMode: '',
        },
        creator: 50000,
        description:
            'This 1-week course aims to provide an introduction to the best practises and tools needed to perform research effectively and reproducibly.\n\nFocusing on solutions around handling biological data, we will cover introductory lessons in R, version control, statistical analyses, and data management. The R component of the course will cover from basic steps in R to how to use some of the most popular R packages (dplyr and ggplot2) for data manipulation and visualisation. No prior R experience or previous knowledge of programming/coding is required. At the end of the course we will address issues relating to reusability and reproducibility.',
        domains: ['Course'],
        provider: 'University of Cambridge',
        title: 'Bioinformatics for Biologists: An introduction to programming, analysis, and reproducibility',
        type: 'course',
        myEntity: false,
    },
    {
        id: 50009,
        award: ['PhD'],
        courseOptions: {
            flexibleDates: true,
            startDate: null,
            studyMode: '',
        },
        creator: 50000,
        description: 'Projects that apply computational approaches to translational clinical (BRC) or biological research (SBS).',
        domains: ['Course'],
        provider: 'EMBL-EBI and Cambridge Biomedical campus',
        title: 'EMBL-EBI EBPOD programme',
        type: 'course',
        myEntity: false,
    },
    {
        id: 50025,
        award: [],
        courseOptions: {
            flexibleDates: true,
            startDate: null,
            studyMode: '',
        },
        creator: 50000,
        description:
            'This workshop will focus on the core steps involved in calling variants with the Broad’s Genome Analysis Toolkit, using the “Best Practices” developed by the GATK team. You will learn why each step is essential to the variant discovery process, what are the operations performed on the data at each step, and how to use the GATK tools to get the most accurate and reliable results out of your dataset.\n\nIn the course of this workshop, we highlight key functionalities such as the germline GVCF workflow for joint variant discovery in cohorts, somatic variant discovery using MuTect2, and copy number variation discovery using GATK-CNV. All analyses are demonstrated using GATK version 4. Finally, we demonstrate the use of pipelining tools to assemble and execute GATK workflows.\nThe workshop covers basic genomics, all currently supported Best Practices pipelines as well as pipelining with WDL/Cromwell/FireCloud. This includes the logic of the major pipelines, file formats and data transformations involved, and hands-on operation of the tools using goal-oriented exercises.',
        domains: ['Course'],
        provider: 'University of Cambridge',
        title: 'Variant Discovery with GATK4',
        type: 'course',
        myEntity: false,
    },
];
