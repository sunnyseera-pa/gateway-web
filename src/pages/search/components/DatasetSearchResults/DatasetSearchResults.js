import React from 'react';
import _ from 'lodash';
import RelatedObject from '../commonComponents/relatedObject/RelatedObject';
import SearchResults from '../commonComponents/SearchResults';

const DatasetSearchResults = ({ updateOnFilterBadge, ...outerProps }) => {
	const mapResults = React.useCallback(
		data => {
			return data.map(dataset => {
				let datasetPublisher;
				let datasetLogo;

				!_.isEmpty(dataset.datasetv2) && _.has(dataset, 'datasetv2.summary.publisher.name')
					? (datasetPublisher = dataset.datasetv2.summary.publisher.name)
					: (datasetPublisher = '');

				!_.isEmpty(dataset.datasetv2) && _.has(dataset, 'datasetv2.summary.publisher.logo')
					? (datasetLogo = dataset.datasetv2.summary.publisher.logo)
					: (datasetLogo = '');

				return (
					<RelatedObject
						key={dataset.id}
						data={dataset}
						activeLink={true}
						onSearchPage={true}
						updateOnFilterBadge={updateOnFilterBadge}
						datasetPublisher={datasetPublisher}
						datasetLogo={datasetLogo}
					/>
				);
			});
		},
		[updateOnFilterBadge]
	);

	return <SearchResults {...outerProps} results={mapResults} />;
};

export default DatasetSearchResults;
