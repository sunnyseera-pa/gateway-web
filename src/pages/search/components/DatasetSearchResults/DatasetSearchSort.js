import React from 'react';
import SortDropdown from '../../../commonComponents/SortDropdown';

const DatasetSearchSort = ({ sort, onSort, search, ...outerProps }) => {
	return (
		<SortDropdown
			onSort={onSort}
			defaultValue={search === '' ? 'metadata' : 'relevance'}
			value={sort}
			options={['relevance', 'popularity', 'metadata', 'latest', 'resources']}
			{...outerProps}
		/>
	);
};

export default DatasetSearchSort;
