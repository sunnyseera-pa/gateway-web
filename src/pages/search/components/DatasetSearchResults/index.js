import DatasetSearchResults from './DatasetSearchResults';

export default DatasetSearchResults;
