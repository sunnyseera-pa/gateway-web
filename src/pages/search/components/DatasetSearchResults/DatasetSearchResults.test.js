import { render } from '@testing-library/react';
import React from 'react';
import DatasetSearchResults from '.';
import { data } from './mockData';

const mockRelatedObject = jest.fn();

jest.mock('../../../commonComponents/relatedObject/RelatedObject', () => props => {
	mockRelatedObject(props);
	return <div />;
});

let wrapper;

const props = {
	data,
	count: 656,
	pageNumber: 3,
	totalPages: 10,
	type: 'datasets',
	search: 'search term',
	updateOnFilterBadge: jest.fn(),
	onPagination: jest.fn(),
	maxResults: 40,
	isLoading: false,
	sort: 'Sort goes here...',
};

describe('Given the DatasetSearchResults component', () => {
	describe('When it is rendered', () => {
		beforeAll(() => {
			wrapper = render(<DatasetSearchResults {...props} />);
		});

		it('Then matches the previous snapshot', () => {
			expect(wrapper.container).toMatchSnapshot();
		});

		it('Then has the correct number of results', () => {
			expect(mockRelatedObject).toHaveBeenCalledTimes(40);
		});

		it('Then passes the correct props', () => {
			expect(mockRelatedObject.mock.calls[0][0]).toEqual({
				data: data[0],
				activeLink: true,
				datasetLogo: '',
				datasetPublisher: 'BREATHE',
				onSearchPage: true,
				updateOnFilterBadge: props.updateOnFilterBadge,
			});
		});
	});
});
