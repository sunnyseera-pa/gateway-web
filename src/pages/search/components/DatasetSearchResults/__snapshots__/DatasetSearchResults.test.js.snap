// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the DatasetSearchResults component When it is rendered Then matches the previous snapshot 1`] = `
<div>
  <div
    style="margin-top: 8px;"
  >
    <div
      class="row"
    >
      <div
        class="text-right save-dropdown"
      >
        Sort goes here...
      </div>
    </div>
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <ul
      class="pagination"
    >
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="#"
          role="button"
        >
          1
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="#"
          role="button"
        >
          2
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="#"
          role="button"
        >
          3
        </a>
      </li>
      <li
        class="page-item active"
      >
        <span
          class="page-link"
        >
          4
          <span
            class="sr-only"
          >
            (current)
          </span>
        </span>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="#"
          role="button"
        >
          5
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="#"
          role="button"
        >
          6
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="#"
          role="button"
        >
          7
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="#"
          role="button"
        >
          8
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="#"
          role="button"
        >
          9
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="#"
          role="button"
        >
          10
        </a>
      </li>
    </ul>
  </div>
</div>
`;
